app/models/user.rb

```diff
@@ -1,21 +1,39 @@
 class User < ApplicationRecord
-  VALID_EMAIL_REGEX = Settings.string_regex
-  before_save {email.downcase!}
+  attr_accessor :remember_token
+  before_save {self .email = email.downcase}
```

Comment:
```suggestion
  before_save {email.downcase!}
```
phân biệt giúp a downcase vs downcase! nhé

Reply:
> phân biệt giúp a downcase vs downcase! nhé

downcase! sẽ trả về nil nếu như xâu ký tự không có thay đổi.
downcase trả về một chuỗi mới và không thay đổi chuỗi hiện có